components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -0,0 +1,438 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.scim2.common.impl;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.json.JSONArray;
+import org.wso2.carbon.identity.role.mgt.core.GroupBasicInfo;
+import org.wso2.carbon.identity.role.mgt.core.IdentityRoleManagementException;
+import org.wso2.carbon.identity.role.mgt.core.RoleBasicInfo;
+import org.wso2.carbon.identity.role.mgt.core.RoleManagementService;
+import org.wso2.carbon.identity.role.mgt.core.UserBasicInfo;
+import org.wso2.carbon.identity.scim2.common.utils.SCIMCommonConstants;
+import org.wso2.carbon.identity.scim2.common.utils.SCIMCommonUtils;
+import org.wso2.charon3.core.attributes.ComplexAttribute;
+import org.wso2.charon3.core.attributes.MultiValuedAttribute;
+import org.wso2.charon3.core.attributes.SimpleAttribute;
+import org.wso2.charon3.core.exceptions.BadRequestException;
+import org.wso2.charon3.core.exceptions.CharonException;
+import org.wso2.charon3.core.exceptions.ConflictException;
+import org.wso2.charon3.core.exceptions.NotImplementedException;
+import org.wso2.charon3.core.extensions.RoleManager;
+import org.wso2.charon3.core.objects.Group;
+import org.wso2.charon3.core.objects.Role;
+import org.wso2.charon3.core.objects.User;
+import org.wso2.charon3.core.utils.codeutils.ExpressionNode;
+import org.wso2.charon3.core.utils.codeutils.Node;
+import org.wso2.charon3.core.utils.codeutils.OperationNode;
+import org.wso2.charon3.core.utils.codeutils.SearchRequest;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+/**
+ * Implementation of the {@link RoleManager} interface.
+ */
+public class SCIMRoleManager implements RoleManager {
+
+    private static final Log log = LogFactory.getLog(SCIMRoleManager.class);
+    private RoleManagementService roleManagementService;
+    private String tenantDomain;
+    private static final String FILTERING_DELIMITER = "*";
+
+    public SCIMRoleManager(RoleManagementService roleManagementService, String tenantDomain) {
+
+        this.roleManagementService = roleManagementService;
+        this.tenantDomain = tenantDomain;
+    }
+
+    @Override
+    public Role createRole(Role role) throws CharonException, ConflictException, BadRequestException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Creating role: " + role.getDisplayName());
+        }
+        try {
+            // Check if the role already exists.
+            if (roleManagementService.isExistingRole(role.getId(), tenantDomain)) {
+                String error = "Role with name: " + role.getDisplayName() + " already exists in the system.";
```

Comment:
shall we log other context info such as tenant domain as well?

Reply:
Fixed and updated the PR.